### DISCUSSION TITLE
耦合度太高

### DISCUSSION THREAD

**Discussion by chunjiu** [zouyaoji/vue-cesium#378] [General]
为啥不做成依赖不那么强的组件

**Comment by chunjiu** [comment]
非常感谢开源，非常辛苦，已Star，期待后面能优下面几点
1. cesium依赖父组件往下传，每个组件都不独立。
2. 依赖vue3特定的API，而这些API在新版文档中已被删除
3. 用的渲染涵数，但渲染涵数除非常特别的场景，大部分下性能没有模版语法高（具体可看vue3官方文档对渲染涵数的说明）

**Reply by zouyaoji** [reply]
主要的局限性是 viewer 获取问题，设计依赖于 vue 的 inject 和 provide 获取全局viewer，所以只有将其他组件都放到vc-viewer下。

如果有更好的设计可以分享。

**Reply by chunjiu** [reply]
为啥不改成Cesium、viewer都由用户向下传递？库、组件只实现功能。
